### PR TITLE
Docs secure app access enhacement

### PR DESCRIPTION
**Pull request by fabianmuehlberger**
Mostly stylistic improvements, including prettier notes, slight rephrasing.
Fixed some typos. 
moved the Cloudflare portion to a separate file and linked to it through the _index.md file, as proposed in my initial commit. 

Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

**Comment by DjP-iX** [comment]
Thanks @fabianmuehlberger.

I'm not sure if I'm understanding the intended structure here, but wouldn't it make more sense to move the #Scope section into the child article, since that tutorial involves configuring Nextcloud?

And then the bottom part of the index file could be something more like 

> Tutorials in this section provide instructions on ...

File: content/SCALE/SCALETutorials/Apps/AppSecurity/CloudflareTunnel.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+A free account is sufficient.
+
+Follow Cloudflare documentation to [register a domain](https://developers.cloudflare.com/registrar/) and set up [DNS](https://developers.cloudflare.com/dns/).
+
+### Creating a Cloudflare Tunnel
+
+[Cloudflare Tunnel](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/) is a client that runs on the TrueNAS SCALE system and proxies traffic from the Cloudflare network to the application.
+
+{{< trueimage src="/images/SCALE/Apps/CloudflareTunnelOverview.jpg" alt="Cloudflare Tunnel Overview" id="Cloudflare Tunnel Overview" >}}
+<p style="text-align: center;"><small> Illustration via <a href="https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/">Cloudflare</a> (CC BY) </small></p>
```
Comment (DjP-iX) [inline content/SCALE/SCALETutorials/Apps/AppSecurity/CloudflareTunnel.md:13]: I ended up adding `caption` support to the trueimage shortcode (thanks for the idea) so you can just use this now:

```suggestion
{{< trueimage src="/images/SCALE/Apps/CloudflareTunnelOverview.jpg" alt="Cloudflare Tunnel Overview" id="Cloudflare Tunnel Overview" caption="Illustration via [Cloudflare](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/) (CC BY)" >}}
```

**Comment by fabianmuehlberger** [comment]
> Thanks @fabianmuehlberger.
> 
> I'm not sure if I'm understanding the intended structure here, but wouldn't it make more sense to move the #Scope section into the child article, since that tutorial involves configuring Nextcloud?
> 
> And then the bottom part of the index file could be something more like 
> 
> > Tutorials in this section provide instructions on ...

You are correct, I am also not quite happy with it. Especially if different apps or methods are added later on. Not sure if the structure: security method > app is the approach. I will have at the other app tutorials to see if this can be improved

**Comment by DjP-iX** [comment]
This is looking good. I reordered the tunnel and Cloudflared app sections for better flow and wrote out a few more incremental steps. 

Before we merge it, I'd like to add one more step at the end of `### Creating a Cloudflare Tunnel` that explains where to find and copy the tunnel token from the Cloudflare UI but I don't have an account to test that. Could you add that in please?

**Comment by fabianmuehlberger** [comment]
> This is looking good. I reordered the tunnel and Cloudflared app sections for better flow and wrote out a few more incremental steps.
> 
> Before we merge it, I'd like to add one more step at the end of `### Creating a Cloudflare Tunnel` that explains where to find and copy the tunnel token from the Cloudflare UI but I don't have an account to test that. Could you add that in please?

Added content as requested.

**Review by DjP-iX — APPROVED** [review]
Looks great, thanks!

**Pull request merged**